yorkie/src/main/kotlin/dev/yorkie/document/crdt/ElementRht.kt

```diff
@@ -0,0 +1,156 @@
+package dev.yorkie.document.crdt
+
+import dev.yorkie.document.time.TimeTicket
+import dev.yorkie.util.YorkieLogger
+
+/**
+ * [ElementRht] is a hashtable with logical clock(Replicated hashtable).
+ */
+internal class ElementRht<T : CrdtElement> : Iterable<ElementRht.Node<T>> {
+    private val logTag = "ElementRhtTag"
+
+    private val nodeMapByKey: MutableMap<String, Node<T>> = mutableMapOf()
+
+    private val nodeMapByCreatedAt: MutableMap<TimeTicket, Node<T>> = mutableMapOf()
+
+    /**
+     * Sets the [value] using the given [key].
+     * If the object exists in [nodeMapByKey] by same [key] then return [CrdtElement], otherwise null.
+     */
+    operator fun set(key: String, value: T): T? {
+        var removed: T? = null
+        val node = nodeMapByKey[key]
+        if (node != null && !node.isRemoved && node.remove(value.createdAt)) {
+            removed = node.value
+        }
+
+        val newNode = Node(key, value)
+        nodeMapByCreatedAt[value.createdAt] = newNode
+        if (node == null || node.value.createdAt < value.createdAt) {
+            nodeMapByKey[key] = newNode
+        }
+        return removed
+    }
+
+    /**
+     * Removes the element in [nodeMapByCreatedAt] using the given key [createdAt],
+     * and [TimeTicket] will be removed by [executedAt].
+     * If the object exists in [nodeMapByCreatedAt] by same key [createdAt], it will return.
+     *
+     * @throws NoSuchElementException if [Node] doesn't exist.
+     */
+    fun remove(createdAt: TimeTicket, executedAt: TimeTicket): T {
+        val node = nodeMapByCreatedAt[createdAt]
+            ?: throw NoSuchElementException("The ElementRhtNode by $createdAt doesn't exist.")
+        node.remove(executedAt)
+        return node.value
```

Comment:
@hackerwins I returned the `node.value` regardless of whether or not the `node.remove()` returned true, based on the JS-SDK.